spec/requests/omniauth_callbacks_controller_spec.rb

```diff
@@ -19,23 +19,19 @@ def request!
   end
 
   context 'when the omniauth setup is returning with an authorization' do
+    # The auth hash data has been observed by connecting to the Keycloak server
+    # https://login.lescommuns.org/.
     before do
       OmniAuth.config.mock_auth[:openid_connect] = OmniAuth::AuthHash.new(
-        'provider' => 'openid_connect',
-        'uid' => 'john@doe.com',
-        'info' => {
-          'email' => 'john@doe.com',
-          'first_name' => 'John',
-          'last_name' => 'Doe'
-        }
+        JSON.parse(file_fixture("omniauth.auth.json").read)
```

Comment:
No, I don't think so. VCR hooks into requests made by us. But this data is sent by the auth server. There may actually be a bit of back and forth under the hood, handled by the omniauth gem. I don't actually know which flow it's using, but I think it's this one: https://developers.onelogin.com/openid-connect

* Our form posts to an omniauth page within our Rails app.
* Omniauth is redirecting the user to lescommuns.
* User is authenticated there, usually performing a login.
* Lescommuns is redirecting back to our omniauth with a token.
* Omniauth issues one or two more requests to obtain user info with that token _(this could be recorded via VCR)_.
* Omniauth calls a method we defined on a controller and gives us the above hash with login data after it did all the authentication work.

If we wanted to record more, we would need the following steps:

* Change test env to use `localhost:3000` instead of `test.host:xxxx`. Our current test hostname and random port is not registered with lescommuns.
* Allow live requests to lescommuns from the browser. Then we should be able to perform a login on lescommuns in a system spec. We need secret env vars to supply username and password for the login.
* But for offline tests we would need to somehow record the incoming request from the browser when it's redirected from lescommuns to our Rails app with a token.
* Instead of navigating live to lescommuns, we would need to emulate that step and redirect to the omniauth controller with our recorded request.
* The rest can be covered by VCR.

My conclusions were:

* too hard for value
* Omniauth has a very convenient test mode which skips all that.
* I recorded the auth hash manually by hooking in with pry but the realistic test setup still needs to record a request to our server. The next step would be to automate just that recording somehow, it would be needed for either solution.
* Given that we have some kind of recording of an incoming request, we might as well just test our interface with omniauth instead of mocking some of its behaviour. We should test our code, not theirs.